src/groucho.js

```diff
@@ -22,10 +22,15 @@ var groucho = window.groucho || {};
         'addons': {}
       };
 
-  // Set empty configs to defaults.
-  for (var config in defaults) {
-    if (!groucho.config.hasOwnProperty(config)) {
```

Comment:
I might adjust your solution slightly...
```javascript
// Handle empty config.
if (!groucho.hasOwnProperty(config) {
  groucho.config = groucho.defaults;
}
else {
  // Set each empty config to the defaults.
  for (var setting in defaults) {
    if (!groucho.config.hasOwnProperty(setting)) {
      groucho.config[setting] = defaults[setting];
    }
  }
}
```
Also, changing the interator variable name for clarity.  Thanks!